packages/flutter_test/lib/src/test_async_utils.dart

```diff
@@ -302,6 +307,10 @@ class TestAsyncUtils {
 
   static _StackEntry? _findResponsibleMethod(StackTrace rawStack, String method, List<DiagnosticsNode> information) {
     assert(method == 'guard' || method == 'guardSync');
+    // Web/JavaScript stack traces use a different format.
```

Comment:
is it possible to implement this for the web? (it should at least be a TODO or open issue, right?)

Reply:
The web stack traces do not have all of the information expected by the diagnostic (class names will be missing, and source lines will reflect compiled JavaScript output instead of Dart source lines)

With this patch the test library will output the "Guarded function conflict" message along with the relevant raw stack traces.

(Without the patch the test library fails an assertion while trying to parse the web stack traces)

Will leave https://github.com/flutter/flutter/issues/107099 open if there is interest in more improvements to the diagnostic for web platforms.

Reply:
sounds good, thanks!